### ISSUE TITLE
Game crash when opening food book with Patchouli installed

### ISSUE THREAD

**Issue by RisingInIris2017** [Cazsius/Spice-of-Life-Carrot-Edition#87]
#### Mod version:
solcarrot-1.15.2-1.9.4

#### Forge version:

1.15.2-31.1.37

#### Singleplayer or Multiplayer:

Singleplayer

#### Installed mods:
Attached in the crash-report.

## Link to full crash log
https://paste.ubuntu.com/p/8phcwbSZ3F/

## Description of the problem
I have edited the `solcarrot-server.toml` config file and move it from `saves\MySave\serverconfig` folder to `.minecraf\defaultconfig` folder. 
And then game crashed every time I open the Food Book.
Then I moved the `solcarrot-server.toml` file to `.minecraft\config` folder,
and created a new save, but the game still crashed when I open that book,

Maybe there is something wrong which was done by myself? 
The link to my edited `solcarrot-server.toml` file:
https://paste.ubuntu.com/p/F44kwcWnc4/

I'm not so familiar to 1.15.2 config editing.
If anything I have done is incorrect, please point it out.
Thanks for your kind help in advance!

**Comment by juliand665** [comment]
Thanks for the detailed report; that's very helpful! Your config looks fine to me; I'll have to see if I can reproduce this locally. I've noticed some crashing issues with the food book myself (playing with Create in 1.14) but couldn't quite track them down.

**Comment by juliand665** [comment]
It doesn't seem like the crash would be affected by the location of your config file at all. It happens when trying to construct the config info page of the book, but more specifically when asking minecraft for the width of its title, which is a static string. I can't really explain this to myself… Which language do you have set in minecraft? That might have something to do with it.

**Comment by RisingInIris2017** [comment]
> Which language do you have set in minecraft?

I am using zh_cn or `中文（中国）`, with Force Unicode Font enabled.

Edited:
I have done two tests:
1. Set language to zh_cn, and disable the Force Unicode Fonts.
I opened a Food Book, and game crashed several seconds later.
2. Set language to en_us, and enable the Force Unicode Fonts.
Game stopped crashing.

So maybe it can be concluded that this issue is related to language settings.

**Comment by juliand665** [comment]
Hm, I couldn't reproduce with zh_cn language and force unicode fonts set to on or off… Are you using any other mods? Maybe even SmoothFont with a custom font—that might do it.

**Comment by RisingInIris2017** [comment]
> Are you using any other mods?

Yes. But I did not find mods like SmoothFonts. I did not even installed Optifine.
Here is my modlist. 
https://paste.ubuntu.com/p/r5fSJvPDqn/

**Comment by juliand665** [comment]
Interesting. If you'd like to help speed this up, you can do for me what I'd resort to otherwise, which is starting out with all those mods and removing mods (or groups of mods) individually to see after which removal the crash stops happening. Then hopefully you'll end up with just SoL: Carrot and that mod, leaving us with a nice distilled incompatibility.

**Comment by RisingInIris2017** [comment]
Sorry for bothering you for so long.
I found the incompatible mods: Only when Kiwi and Patchouli are both installed will this issue occur.
New crash-report is here: https://paste.ubuntu.com/p/s6P6XJysGz/
I will report this issue to Kiwi and Patchouli's devs.

**Comment by RisingInIris2017** [comment]
Now maybe I can confirm that SoL and Patchouli have incompatibility issue.
https://paste.ubuntu.com/p/WdzXS74CJh/

**Comment by juliand665** [comment]
Thanks for the detailed investigation! I wonder what the Patchouli people will have to say about it.

**Comment by RisingInIris2017** [comment]
I have tried the latest Patchouli, but game still crashes.
New crash reports is attached here:
https://github.com/Vazkii/Patchouli/issues/216#issuecomment-621574852

**Comment by HelloCard** [comment]
still happened at solcarrot-1.21.1-1.16.3 and Patchouli-1.21-87-NEOFORGE only:
https://pastebin.com/m9qF2fEE
